packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -457,6 +448,54 @@ const updateSearchIndexSucceeded = (): UpdateSearchIndexSucceededAction => ({
   type: ActionTypes.UpdateSearchIndexSucceeded,
 });
 
+export const POLLING_INTERVAL = 5000;
+
+let pollInterval: ReturnType<typeof setInterval> | undefined = undefined;
+
+export const openSearchIndexes = (): IndexesThunkAction<
+  Promise<void>,
+  SearchIndexesOpenedAction | FetchSearchIndexesActions
+> => {
+  return async function (dispatch, getState) {
+    if (getState().searchIndexes.isVisible || pollInterval) {
+      return;
+    }
+
+    dispatch({
+      type: ActionTypes.SearchIndexesOpened,
+    });
+
+    pollInterval = setInterval(() => {
+      void dispatch(pollSearchIndexes());
+    }, POLLING_INTERVAL);
+
+    // if this is the first time, also refresh the list
+    if (getState().searchIndexes.status === 'NOT_READY') {
+      await dispatch(refreshSearchIndexes());
+    }
```

Comment:
I can move it. For some reason I just had it in my mind that I shouldn't fire off actions in there.

Reply:
If they are state setters we should avoid in favor of just using initial state, but otherwise activate is part of the lifecycle similar to react lifecycles, so similarly actions can generally happen in those lifecycles